src/components/ShareTrajectoryModal/EmbedSnippetPanel.tsx

```diff
@@ -0,0 +1,133 @@
+import React from "react";
+import classNames from "classnames";
+import { Input, Checkbox, Button } from "antd";
+import { URL_PARAM_KEY_FILE_NAME } from "../../constants";
+import VerticalFlexbox from "../../styles/utils";
+import { getUrlParamValue, copyToClipboard } from "../../util";
+import { DownArrow } from "../Icons";
+
+import styles from "./style.css";
+
+const DEFAULT_WIDTH = 834;
+const DEFAULT_HEIGHT = 480;
+const MAX_WIDTH = 3840;
+const MAX_HEIGHT = 2160;
+
+interface EmbedSnippetPanelProps {
+    startTime: number;
+}
+
+const EmbedSnippetPanel = ({ startTime }: EmbedSnippetPanelProps) => {
+    const trajectory = getUrlParamValue(
+        window.location.href,
+        URL_PARAM_KEY_FILE_NAME
+    );
+
+    const [width, setWidth] = React.useState(DEFAULT_WIDTH);
+    const [height, setHeight] = React.useState(DEFAULT_HEIGHT);
+    const [constrainProportions, setConstrainProportions] =
+        React.useState(false);
```

Comment:
This is the one thing I feel a bit confused on. I think I understand what you've done here, basically maintaining the current aspect ratio whenever a new height or width changes. Is that what UX wanted for this?

I took it to mean allowing the user to provide height and width at all, rather than just using the default, which I guess was a misunderstanding.

I find the phrase constrain proportions a bit misleading in both cases. Maintain aspect ratio, or allow size inputs or something like that would be clearer imo.

Reply:
I feel like that's the wording for photoshop, but yeah, I'm 99% certain this is the functionality she meant. if you think the wording is confusing you can ask UX for a change

Reply:
I'll take your word for it, thanks for the fix.